prov/rxm/src/rxm_ep.c

```diff
@@ -1360,7 +1360,10 @@ rxm_ep_send_common(struct rxm_ep *rxm_ep, struct rxm_conn *rxm_conn,
 				rxm_ep_do_progress(&rxm_ep->util_ep);
 			rxm_tx_buf_release(rxm_ep, RXM_BUF_POOL_TX, tx_buf);
 		}
-	} else if (data_len <= rxm_ep->sar_limit) {
+	} else if (data_len <= rxm_ep->sar_limit &&
+		   /* SAR uses eager_limit as segment size */
+		   rxm_ep->eager_limit <= sizeof_field(struct ofi_ctrl_hdr,
+						       seg_size)) {
```

Comment:
Is eager_limit specified in number of bits?

Reply:
Oops. That's an error, will fix it.